### ISSUE TITLE
Fix only specified offense

### ISSUE THREAD

**Issue by stdedos** [rubocop-hq/rubocop#6544]
**Is your feature request related to a problem? Please describe.**

I would like to be able to fix a very specific offense.
e.g.
```ruby
require 'socket'
addr_infos = Socket.getifaddrs
addr_infos.each do |addr_info|
  puts(addr_info.ai)
  if addr_info.addr
    puts("#{addr_info.name} has address #{addr_info.addr.ip_address}")
  end
end
```

```ruby
$  rubocop -V
0.60.0 (using Parser 2.5.3.0, running on ruby 2.4.0 x86_64-linux)
#[...]
-:623:11: C: Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
          if addr_info.addr
          ^^

```
Using Rubocop though RubyMine, I can get the option to `Fix (all) [Style/IfUnlessModifier] Rubocop offences`
`_____________ Implied; not written; added for clarity ^^^^^^`

_and I assume there is a "way" to run that in `rubocop` directly. [1]_

**Describe the solution you'd like**

I would like something along the lines of:
`rubocop --fix-line=623 filename` that would fix, if found, error(s) in that line.

The same "filtering" mechanism [^1] can be used in conjuction with this, _maybe_

Other "valid" solutions (within the scope of my work) would be to accept line ranges, or, on VCS-backed projects, _fix only modified code_

**Describe alternatives you've considered**

- Fix it myself
- Fork it to a file, get the auto-fixed-output, patch back

**Additional context**

I understand that the very specific `--fix-line=623` feature has limited usage, but it sounds "easier" than the "fix VCS-changed" lines, as an MVP.
The overall usabilty is, obviously, to be able to minimize scope of changes e.g., I don't want to change legacy code for `xyz` reason, but I do want for Rubocop to fix whatever I will change anyway

**Comment by jaredbeck** [comment]
> [I want] to minimize scope of changes e.g., I don't want to change legacy code .. but I do want for Rubocop to fix whatever I will change anyway

In my experience, this is an uncommon, but legitimate concern.

> Fork it to a file, get the auto-fixed-output, patch back

Yes, or: 

1. auto-correct the whole file in-place and then use `git add --patch` to select only the patch you want, discard the rest with `git checkout`, or
2. disable rubocop in the parts of the file you are afraid to touch, using [`rubocop:disable` comments](http://docs.rubocop.org/en/latest/configuration/#disabling-cops-within-source-code)

Given that this use-case is uncommon, and there are many ways to work around it, I'd suggest that the decision of whether to add this feature or not be based on how much complexity it would add to RuboCop.

**Comment by stdedos** [comment]
Not for this case specifically, but for incrementally fixing a codebase - https://github.com/reviewdog/reviewdog can prove very helpful
